### PR TITLE
PR 1120 for some structure

### PR DESCRIPTION
**Pull request by NPC1314**
## About The Pull Request

Currently its all a pile of random palettes and whims in a huge mess.
Structured it, point is to make colored stuff be not too dark, not too bright, fit in with saturation of other stuff, and not ruin sprites as much. 

* Palette enforced for colorables. Basically just a white palette so applied colors looks reasonably correct in shade and hue
Colorable Chaperon hat most affected, tunic second most. Majority colorables still use their old palette because I based the new one on the most commonly used old informal one.

* Dye bins split into two, cheap and luxury. Cheap can be made with a small log, ash, clod of dirt, a jacksberry and a swampweed, this one gives a selection of greens, browns, yelllows and some other shades, mostly somewhat muted.
Luxury can be made with fyrtitius flowers, sunflowers, a troll horn and turnips, or bought from merchant, and gives access to more vibrant and luxury colors, purple, deep red etc.

* Buckled shoes werent in use, fixed the onmob sprite and added to mason, blacksmith, underdweller
* Courtier hat wasnt in use, added to tailor job (is that even in use tho...)

* Repathed swampweed, jacksberries, pipeweed, dye bin because they had weird pathing
* Pipe weed named pipe weed again with the westleach name as an altenrative in description. Was never meant to change, just carried over from Ratwood.

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

### CODE REVIEW

**Comment by NPC1314** [comment]
Now they can!

File: modular/Neu_Farming/code/produce.dm [hunk 1]
```diff
@@ -263,9 +263,9 @@
 	dust_result = /obj/item/alch/tobaccodust
 	sellprice = 0 // only dried has value
 
-/obj/item/reagent_containers/food/snacks/produce/rogue/dry_pipeweed
+/obj/item/reagent_containers/food/snacks/produce/dry_pipeweed
 	seed = null
-	name = "westleach leaf"
+	name = "dries pipeweed"
```
Comment (ThePainkiller) [inline modular/Neu_Farming/code/produce.dm:268]: You mean, dried?

Also I hate this change, "pipeweed" sounds generic as fuck, while westleach feels more unique

Comment (NPC1314) [reply modular/Neu_Farming/code/produce.dm:268]: I dont much care either way tbh its just the path name, but yeah, dried is what its supposed to be.
Could repath it to westleach instead I guess, not like its any more work really

**Comment by YuiY1997** [comment]
Does this add dye-bins to the map at all? IT might be nice to give the merchant a fancy one roundstart so they can charge for dying purposes.

**Comment by lavillastrangiato** [comment]
Definitely add the dye bin somewhere to the town.
*Do not force people to kill a troll to make a dye bin*.
Lower the asking price. 380 is much too expensive.

**Comment by NPC1314** [comment]
There could be some added to mapping, best would probably be a cheap one in a place you want to get people to congregate to,  but no need to rush that, lets tweak the color palettes and recipes a bit first.

**Comment by anything86** [comment]
I understand the move to make dyes a commodity one might sell in a player economy by making their acquisition harder. At the same time, I think which things get commodified should be carefully considered on a basis of player count and mechanical utility, otherwise you wind up in a situation where there is no one to provide said commodity as there are better, more useful things to sell that require less or equal effort, and it gets consequently ignored.

Having dyes be a single small log to craft was really enjoyable and added some much needed customization without feeling like it slowed down or interfered with your round. If they become a "maybe this round we'll have it, maybe we won't" I think the playerbase as a whole stands to lose something here. With this testmerged, I've seen such rounds where there's simply no expensive dye bin, and even the inexpensive ones are not in sight if even present.

Would adding premapped dye bins alleviate this issue greatly? Yes. Though I also don't look forward to long queues at the merchant just to dye your clothes, for example. I also think this will quickly bring to light the **current bug where items disappear if a second item is inserted into the bin while it's inside**.

**Comment by NPC1314** [comment]
Point of having a recipe beyond a small log is to make it vibe with all the other crafting , having it make at least a modicum of sense, if convenience was all that mattered it could just be a single stick, or a magic summon spell for instant dye bin really, supply pod style.

Anyways, a premapped dye bin isnt a issue, expensive dye bin is meant to be a luxury item and something to spend the mammon on, court has thousands of mammon at its disposal, if they want to they can afford it easily. Or the merchant can buy one and make people pay premium to dye.

 "But my drip" you might say, well that goes for clothing items too, really armor and weapons too, yet some cost more money than others, you dont got every clothing piece available for free round start etc. You can´t have RGB hair colors either on demand, lots of stuff like that. The updated cheap lineup covers a lot of that anyways, it takes some iterations to get all the bits in place with a new system.
 
 Point is this is a system, not just some PR about the specific cost of a dye bin, thats almost irrelevant in the big picture, and trivial to adjust. I'll see which PR can slide in a dye bin in the square, until then you can make them pretty easy still, with swampweed requirement gone.

**Comment by anything86** [comment]
Not at all opposed to a middle ground that feels more realistic than single small log, I should've added that.

**Comment by NPC1314** [comment]
> Not at all opposed to a middle ground that feels more realistic than single small log, I should've added that.

I getcha. If you feel its still expensive or should use other ingridients you can always PR up a edit whenvever, just making clear its the actual system of palettes and hues thats the core of this, the recipes are just a shot from the hip, point isnt got gate people from having fun colors, but its to put some of the more extreme choices behind a social gate of a kind, so visuals will help reinforce themes of feudalism.

 If you see someone in bright I dunno, pink, it should tell you ohh this is a noble or someone with money probably without words. Exactly which hues are put in that category is also up for debate, but thats the underlying logic of it, reinforcing the setting with mechancis subtly

**Comment by sanshoom** [comment]
Somehow I managed to miss this when I first glanced this PR, but holy shit: Why are you changing dye bins.

The dye bin, something that is never abused, rarely used, and is a literal fucking COLORBIN having their recipe changed for no fucking reason. Please god, revert that shit. It's almost as if you coders hate something that can justifiably be simple and fun to use. Oh, reminder, rounds are usually two hours. People don't want to have to spend an absurd amount of time to find shit to craft a better dye bin, when they could alternatively be using that time to ROLEPLAY.

**Comment by NPC1314** [comment]
Now its possible to slapcraft a cheap dye bin by adding just one of the ingridients to a existing wood bin. Can also be kicked over to spill out. Also the double item bug fixed. And there exists the option to craft one from dirt, log and a berry. If thats still too hard may God have mercy on your soul.

**Review by Optimism333 — APPROVED** [review]

**Review by Optimism333 — COMMENTED** [review]
Let me know if this one is ready to go in.

**Pull request merged**